test_conformance/events/test_callbacks.cpp

```diff
@@ -372,3 +383,75 @@ int test_callbacks_simultaneous(cl_device_id deviceID, cl_context context,
     if (actionEvents) delete[] actionEvents;
     return -1;
 }
+
+int test_callback_not_called_simple(cl_device_id deviceID, cl_context context,
+                                    cl_command_queue queue, int num_elements)
+{
+    cl_int error = CL_SUCCESS;
+    clEventWrapper user_event = clCreateUserEvent(context, &error);
+    test_error(error, "clCreateUserEvent failed");
+
+    bool confirmation = false;
+    error = clSetEventCallback(user_event, CL_COMPLETE,
+                               combuf_event_callback_function, &confirmation);
+    test_error(error, "clSetEventCallback failed");
+
+    error = clSetUserEventStatus(user_event, CL_INVALID_VALUE);
+    test_error(error, "clSetUserEventStatus failed");
+
+    // Wait for callback
+    std::this_thread::sleep_for(std::chrono::milliseconds(500));
+
+    if (!confirmation)
+    {
+        log_error("callback not called upon error status different than "
+                  "CL_SUCCESS\n");
+        return TEST_FAIL;
+    }
+
+    return CL_SUCCESS;
+}
+
+int test_callback_not_called_enqueue_command(cl_device_id deviceID,
+                                             cl_context context,
+                                             cl_command_queue queue,
+                                             int num_elements)
+{
+    cl_int error = CL_SUCCESS;
+
+    clEventWrapper user_event = clCreateUserEvent(context, &error);
+    test_error(error, "clCreateUserEvent failed");
+
+    const cl_int pattern_pri = 0xA;
+    clMemWrapper in_mem =
+        clCreateBuffer(context, CL_MEM_READ_ONLY, sizeof(cl_int) * num_elements,
+                       nullptr, &error);
+    test_error(error, "clCreateBuffer failed");
+
+    bool confirmation = false;
+    error = clSetEventCallback(user_event, CL_COMPLETE,
+                               combuf_event_callback_function, &confirmation);
+    test_error(error, "clSetEventCallback failed");
```

Comment:
Since this test is putting the callback on the user event it really isn't doing much different than the "simple" test, above.

I would suggest:

1. Including an event on the fill buffer command, below.
    * Note, the fill buffer command should still depend on the user event, also.
3. Set the callback on the fill buffer event, not on the user event.
4. After setting the user event to an error status the error status should propagate to the fill buffer command (because it is dependent on the user event), and the callback should therefore be called.

Reply:
@bashbaug done